api/v1alpha1/nonadminbackupstoragelocationrequest_types.go

```diff
@@ -0,0 +1,105 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	velerov1 "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NonAdminBSLRequest controls the approval of the NonAdminBackupStorageLocation
+// +kubebuilder:validation:Enum=approve;reject;pending
+type NonAdminBSLRequest string
+
+// NonAdminBSLRequestPhase is the phase of the NonAdminBackupStorageLocationRequest
+// +kubebuilder:validation:Enum=Pending;Approved;Rejected
+type NonAdminBSLRequestPhase string
+
+// Predefined NonAdminBSLRequestConditions
+const (
+	NonAdminBSLRequestApproved NonAdminBSLRequest = "approve"
+	NonAdminBSLRequestRejected NonAdminBSLRequest = "reject"
+	NonAdminBSLRequestPending  NonAdminBSLRequest = "pending"
+)
+
+// Predefined NonAdminBSLRequestPhases
+const (
+	NonAdminBSLRequestPhasePending  NonAdminBSLRequestPhase = "Pending"
+	NonAdminBSLRequestPhaseApproved NonAdminBSLRequestPhase = "Approved"
+	NonAdminBSLRequestPhaseRejected NonAdminBSLRequestPhase = "Rejected"
+)
+
+// NonAdminBackupStorageLocationRequestSpec defines the desired state of NonAdminBackupStorageLocationRequest
+type NonAdminBackupStorageLocationRequestSpec struct {
+	// approvalDecision is the decision of the cluster admin on the Requested NonAdminBackupStorageLocation creation.
+	// +optional
+	ApprovalDecision NonAdminBSLRequest `json:"approvalDecision,omitempty"`
+}
+
+// NonAdminBackupStorageLocationRequestStatusInfo contains information of the related NonAdminBackupStorageLocation object.
+type NonAdminBackupStorageLocationRequestStatusInfo struct {
+	// requestedSpec contains the requested Velero BackupStorageLocation spec from the NonAdminBackupStorageLocation
+	// +optionl
+	RequestedSpec *velerov1.BackupStorageLocationSpec `json:"requestedSpec"`
+
+	// nacuuid references the NonAdminBackupStorageLocation object by it's label containing same NACUUID.
+	// +optional
+	NACUUID string `json:"nacuuid,omitempty"`
+
+	// name references the NonAdminBackupStorageLocation object by it's name.
+	// +optional
+	Name string `json:"name,omitempty"`
+
+	// namespace references the Namespace in which NonAdminBackupStorageLocation exists.
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
+}
+
+// NonAdminBackupStorageLocationRequestStatus defines the observed state of NonAdminBackupStorageLocationRequest
+type NonAdminBackupStorageLocationRequestStatus struct {
+	// +optional
+	NonAdminBackupStorageLocationRequestStatusInfo *NonAdminBackupStorageLocationRequestStatusInfo `json:"nonAdminBackupStorageLocation,omitempty"`
```

Comment:
I though we wanted this as `nonAdminBackupStorageLocation` ?

Reply:
Including structure of `NonAdminBackupStorageLocation` will look wird imo, because it contains yet another status which contains Velero BSL:
https://github.com/migtools/oadp-non-admin/blob/master/api/v1alpha1/nonadminbackupstoragelocation_types.go#L40

Reply:
That's just go var name, which isn't used in the api/v1alpha1, right? If I used the `NonAdminBackupStorageLocation` then we had this object already https://github.com/migtools/oadp-non-admin/blob/121987f57ecb24d4bc65e4a2bef81593525a5a64/api/v1alpha1/nonadminbackupstoragelocation_types.go#L74